source/s3_client.c

```diff
@@ -693,11 +722,18 @@ struct aws_s3_meta_request *aws_s3_client_make_meta_request(
                 error_occurred = true;
                 goto unlock;
             }
-            endpoint->handled_by_client = true;
+            struct aws_task *async_release_task = aws_mem_calloc(client->allocator, 1, sizeof(struct aws_task));
+            aws_task_init(async_release_task, s_endpoint_release_task, endpoint, "se_endpoint_refcount_task");
+
+            endpoint->async_release.client = client;
+            endpoint->async_release.task = async_release_task;
+            endpoint->async_release.task_scheduled = false;
```

Comment:
trivial: should this endpoint initialization code go in in s3_endpoint.c instead of s3_client.c?